securesystemslib/signer/_sigstore_signer.py

```diff
@@ -0,0 +1,156 @@
+                raise ValueError(
+                    f"{content} string required for scheme {scheme}"
+                )
+
+        return cls(keyid, keytype, scheme, keyval, key_dict)
+
+    def to_dict(self) -> Dict:
+        return {
+            "keytype": self.keytype,
+            "scheme": self.scheme,
+            "keyval": self.keyval,
+            **self.unrecognized_fields,
+        }
+
+    def verify_signature(self, signature: Signature, data: bytes) -> None:
+        from sigstore.verify import VerificationMaterials, Verifier
+        from sigstore.verify.policy import Identity
+        from sigstore_protobuf_specs.dev.sigstore.bundle.v1 import Bundle
+
+        verifier = Verifier.production()
+        identity = Identity(
+            identity=self.keyval["identity"], issuer=self.keyval["issuer"]
+        )
+
+        signature_bundle = Bundle().from_dict(signature.signature)
+        verification_materials = VerificationMaterials.from_bundle(
+            input_=io.BytesIO(data), bundle=signature_bundle, offline=True
+        )
+
+        try:
+            result = verifier.verify(verification_materials, identity)
+            if not result:
+                logger.info(
+                    "Key %s failed to verify sig: %s", self.keyid, result.reason
+                )
+                raise exceptions.UnverifiedSignatureError(
+                    f"Failed to verify signature by {self.keyid}"
+                )
+        except exceptions.UnverifiedSignatureError:
+            raise
+
+        except Exception as e:
+            logger.info("Key %s failed to verify sig: %s", self.keyid, str(e))
+            raise exceptions.VerificationError(
+                f"Unknown failure to verify signature by {self.keyid}"
+            ) from e
+
+
+class SigstoreSigner(Signer):
+    """Sigstore signer."""
+
+    def __init__(self, token: str, public_key: Key):
+        # TODO: Vet public key
+        # - signer eligible for keytype/scheme?
+        # - token matches identity/issuer?
+        self.public_key = public_key
+        self._token = token
+
+    @classmethod
+    def from_priv_key_uri(
+        cls,
+        priv_key_uri: str,
+        public_key: Key,
+        secrets_handler: Optional[SecretsHandler] = None,
+    ) -> "SigstoreSigner":
+        raise NotImplementedError()
+
+    def sign(self, payload: bytes) -> Signature:
+        """Signs payload using the OIDC token on the signer instance.
+
+        Arguments:
+            payload: bytes to be signed.
+
+        Raises:
+            Various errors from sigstore-python.
+
+        Returns:
+            Signature.
+
+            NOTE: The ``signature`` attribute of the returned object
+            contains the ``dict`` representation of a sigstore ``Bundle`.
+            This is incompatible with the TUF specification and the
+            ``Signature` interface, which expect the attribute to be of type `str`.
```

Comment:
Could store the bundle in a separate "bundle" field in the signature, nothing forces us to use "sig", right?

To keep spec conformance at 100% could even fill "sig" with the actual signature value -- even if it's useless without the bundle